admin/src/components/util/list/list.jsx

```diff
@@ -0,0 +1,254 @@
+    path: sortParams || "name",
+    order: orderParams || "asc",
+  });
+  const pageSize = 10;
+  const [currentPage, setCurrentPage] = useState(
+    parseInt(pageParams, 10) ? parseInt(pageParams, 10) : 1
+  );
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [showConsolidateModal, setShowConsolidateModal] = useState(false);
+
+  /**
+   * @param {string} newSearchText
+   * Updates the search text state and url.
+   */
+  function handleSearch(newSearchText) {
+    setCurrentPage(1);
+    setSearchText(newSearchText);
+  }
+
+  /**
+   * @param {Array} items
+   * The items to paginate.
+   * @param {number} pageNumber
+   * The chosen page.
+   * @param {number} sizeOfPage
+   * The page size
+   * @returns {Array}
+   * The paginated items.
+   */
+  function paginate(items, pageNumber, sizeOfPage) {
+    const startIndex = (pageNumber - 1) * sizeOfPage;
+    return items.slice(startIndex, startIndex + sizeOfPage);
+  }
+
+  /**
+   * If they search or filter, the pagination is reset.
+   */
+  useEffect(() => {
+    const params = new URLSearchParams();
+    if (searchText) {
+      params.append("search", searchText);
+    }
+    params.append("sort", sortBy.path);
+    params.append("order", sortBy.order);
+    params.append("page", currentPage);
+    history.replace({ search: params.toString() });
+  }, [searchText, sortBy, currentPage]);
+
+  /**
+   * Closes delete modal.
+   */
+  function onCloseDeleteModal() {
+    setShowDeleteModal(false);
+  }
+
+  /**
+   * Closes consolidate modal.
+   */
+  function onCloseConsolidateModal() {
+    setShowConsolidateModal(false);
+  }
+
+  /**
+   * @param {number} page
+   * Updates pagination page.
+   */
+  function handlePageChange(page) {
+    setCurrentPage(page);
+  }
+
+  /**
+   * @param {object} sortColumn
+   * Updates sortcolumn.
+   */
+  function handleSort(sortColumn) {
+    setCurrentPage(1);
+    setSortBy(sortColumn);
+  }
+
+  /**
+   * @param {object} dataToFilter
+   * Search filter function.
+   * @returns {boolean}
+   * Whether the searchtext is in the data entry.
+   */
+  function filterDataFromSearchInput(dataToFilter) {
+    const dataValuesString = Object.values(dataToFilter).join(" ");
+    return dataValuesString
+      .toLocaleLowerCase()
+      .includes(searchText.toLocaleLowerCase());
+  }
+  /**
+   * @param {string|number} a
+   * sort parameter a
+   * @param {string|number} b
+   * sort parameter b
+   * @returns {number}
+   * Sorting number.
+   */
+  function sortData(a, b) {
+    let sortVarA = a[sortBy.path];
+    let sortVarB = b[sortBy.path];
+
+    sortVarA =
+      typeof sortVarA === "string" ? sortVarA.toLocaleLowerCase() : sortVarA;
+    sortVarB =
+      typeof sortVarB === "string" ? sortVarB.toLocaleLowerCase() : sortVarB;
+    if (sortVarA < sortVarB) {
+      return -1;
+    }
+    if (sortVarA > sortVarB) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  /**
+   * @returns {object}
```

Comment:
Missing function description, please check all code and add missing descriptions.